async-http-client/src/main/scala/org/http4s/client/asynchttpclient/AsyncHttpClient.scala

```diff
@@ -122,9 +126,11 @@ object AsyncHttpClient {
       override def onThrowable(throwable: Throwable): Unit =
         invokeCallback(logger)(cb(Left(throwable)))
 
-      // it's okay to invoke this repeatedly since repeated async callbacks are dropped by law
       override def onCompleted(): Unit =
-        invokeCallback(logger)(cb(Right(response -> dispose)))
+        onStreamCalled.get
+          .ifM(F.unit, F.delay(invokeCallback(logger)(cb(Right(response -> dispose)))))
```

Comment:
`F.delay`ing something that calls `unsafeRunSync` to be called in an `unsafeRunSync` feels abusive, even for a spot we need `Effect`. I'd like to think harder about this, but more importantly, I'd like to merge the bugfix.

Reply:
This is also one of the places where using the `ifTrue` and `ifFalse` labels helps readability a lot.